test/plugin_functional/test_suites/core_plugins/application_deep_links.ts

```diff
@@ -8,20 +8,24 @@
 
 import url from 'url';
 import expect from '@kbn/expect';
-import { PluginFunctionalProviderContext } from '../../services';
+import type { PluginFunctionalProviderContext } from '../../services';
 
 export default function ({ getService, getPageObjects }: PluginFunctionalProviderContext) {
   const PageObjects = getPageObjects(['common']);
-
   const browser = getService('browser');
   const appsMenu = getService('appsMenu');
   const testSubjects = getService('testSubjects');
   const retry = getService('retry');
   const esArchiver = getService('esArchiver');
   const log = getService('log');
 
-  const loadingScreenNotShown = async () =>
-    expect(await testSubjects.exists('kbnLoadingMessage')).to.be(false);
+  const clickAppLink = async (app: string) => {
+    const appLink = `dlNav${app}`;
+    if (!(await testSubjects.exists(appLink))) {
+      log.debug(`App ${app} not found on side nav`);
+    }
+    await testSubjects.click(appLink);
+  };
```

Comment:
Super cool! I'd be great to have this in the helpers.

Reply:
This one is custom to the _deep links_ apps (note the `dlNav`), but perhaps we can generalise it somehow.
In fact, I have a few more improvements in the pipes for other tests in the `test_suites/core_plugins`. Stay tunned!